### PR TITLE
Hotfix - #113 testing/correcting hashing for windows file path length limitations

### PR DESCRIPTION
**Pull request by Ocramius**
Provides tests for #113

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Cache/FileCacheTest.php [hunk 1]
```diff
@@ -129,4 +129,50 @@ public function testGetExtensionReturnsExtensionString()
...
+        $getFileName = new \ReflectionMethod($fileCache, 'getFilename');
+
+        $getFileName->setAccessible(true);
+
+        $this->assertSame(
+            __DIR__ . '/' . substr($tooLongKeyHash, 0, 2) . '/_' . $tooLongKeyHash . '.doctrine.cache',
+            $getFileName->invoke($fileCache, $tooLongKey),
+            'Keys over the limit of the allowed length are hashed correctly'
+        );
+        $this->assertSame(
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/FileCacheTest.php]: Meh, needs some hardening around this assertion (fails on travis).

The annoying bit is that the number of chars in a hash is always even, while the sum of extension + base-path can be odd: fun!

File: lib/Doctrine/Common/Cache/FileCache.php [hunk 2]
```diff
@@ -137,7 +137,7 @@ protected function getFilename($id)
         if (
             '' === $id
             || ((strlen($id) * 2 + $this->extensionStringLength) > 255)
-            || (($this->isRunningOnWindows && $this->directoryStringLength + 4 + strlen($id) * 2 + $this->extensionStringLength) > 259)
+            || ($this->isRunningOnWindows && ($this->directoryStringLength + 4 + strlen($id) * 2 + $this->extensionStringLength) > 259)
```
Comment (zeroedin-bill) [inline lib/Doctrine/Common/Cache/FileCache.php]: This hurts my eyes.

File: tests/Doctrine/Tests/Common/Cache/FileCacheTest.php [hunk 3]
```diff
@@ -129,4 +129,50 @@ public function testGetExtensionReturnsExtensionString()
...
+            array(__DIR__, '.doctrine.cache')
+        );
+
+
+        $baseDirLength        = strlen(__DIR__);
+        $extensionLength      = strlen('.doctrine.cache');
+        $windowsPathMaxLength = 260;
+        $maxKeyLength         = $windowsPathMaxLength - ($baseDirLength + $extensionLength);
+
+        self::assertSame('61', bin2hex('a'), '(added just for clarity and system integrity check)');
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/Common/Cache/FileCacheTest.php]: Where is this going to fail? SPARC?

File: tests/Doctrine/Tests/Common/Cache/FileCacheTest.php [hunk 4]
```diff
@@ -129,4 +129,50 @@ public function testGetExtensionReturnsExtensionString()
...
+     */
+    public function testWindowsPathLengthLimitationsAreCorrectlyRespected()
+    {
+        if (! defined('PHP_WINDOWS_VERSION_BUILD')) {
+            define('PHP_WINDOWS_VERSION_BUILD', 'Yes, this is the "usual suspect", with the usual limitations');
+        }
+
+        $fileCache = $this->getMockForAbstractClass(
+            'Doctrine\Common\Cache\FileCache',
+            array(__DIR__, '.doctrine.cache')
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/Common/Cache/FileCacheTest.php]: I don't see the need to use __DIR__ here - better to use sys_get_temp_dir() IMO

File: tests/Doctrine/Tests/Common/Cache/FileCacheTest.php [hunk 5]
```diff
@@ -129,4 +129,50 @@ public function testGetExtensionReturnsExtensionString()
...
+        );
+
+
+        $baseDirLength        = strlen(__DIR__);
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/Common/Cache/FileCacheTest.php]: same RE: __DIR__

File: tests/Doctrine/Tests/Common/Cache/FileCacheTest.php [hunk 6]
```diff
@@ -129,4 +129,50 @@ public function testGetExtensionReturnsExtensionString()
...
+        $extensionLength      = strlen('.doctrine.cache');
+        $windowsPathMaxLength = 260;
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/Common/Cache/FileCacheTest.php]: The 260 includes the terminating null. So in php terms this should be 259.

**Pull request merged**